src/contrib/ethosu/cascader/stripe_config.h

```diff
@@ -0,0 +1,218 @@
+   * then axis 2, then finally axis 1.
+   */
+  inline std::vector<int> GetOrder() const { return order_; }
+  /*!
+   * \brief Get the stripes of the stripe config.
+   * \return The stripes of the stripe config.
+   * \note The stripes refer to the number of stripes in each axis.
+   * There must be at least one stripe in any given axis.
+   */
+  inline std::vector<int> GetStripes() const { return stripes_; }
+  /*!
+   * \brief Get the offset of the stripe config.
+   * \return The offset of the stripe config.
+   * \note The offset refers to the offset of the first stripe
+   * from the first element of the tensor. For example, in a 2D padding operation
+   * that is padding by 1 in every dimension, the offset would be [-1, -1].
+   */
+  inline std::vector<int> GetOffset() const { return offset_; }
+  /*! \return The hash of the StripeConfigNode */
+  size_t GetHash() const { return hash_; }
+
+  static constexpr const char* _type_key = "contrib.ethosu.cascader.StripeConfig";
+  TVM_DECLARE_FINAL_OBJECT_INFO(StripeConfigNode, Object);
+
+ protected:
+  friend class StripeConfig;
+  friend class PropagatorNode;
+
+  /*! \brief Compute the hash of the StripeConfigNode */
+  void ComputeHash_();
+
+  /*! \brief The shape of the stripes */
+  std::vector<int> shape_;
+  /*! \brief The extent of region to stripe over */
+  std::vector<int> extent_;
+  /*! \brief The strides of the stripes */
+  std::vector<float> strides_;
+  /*! \brief The order of the striping axes */
+  std::vector<int> order_;
+  /*! \brief The number of stripes in each axis */
+  std::vector<int> stripes_;
+  /*! \brief The offset of the first stripe */
+  std::vector<int> offset_;
+  /*! \brief The hash of the StripeConfigNode */
+  std::size_t hash_{0};
+};
+
+/*!
+ * \brief An object to describe how a tensor should be computed as a series
+ of n-dimensional tiles, or 'stripes'.
+ * \note The StripeConfig is a verbose way of specifying how to tile a tensor.
+ * We can imagine taking a 2D tensor of size (12, 12) and wanting to compute
+ * it in tiles of (4, 4). The tile is referred to as a stripe here to generalize
+ * this to n-dimensional tiles.
+ *
+ * The size of that stripe in each axis is the 'shape'. The strides is how far
+ * you should move between stripes, so also (4, 4) for a simple non-overlappping
+ * tiling. However, we explore some overlapping scheduling options so shape != strides
+ * in general. The 'extent' is simply (12, 12), the region over which we're conducting
+ * our tiling.
+ *
+ * The 'order' tells us which axis to iterate over first and which second and the
+ * 'stripes' tells us how many stripes we need to compute in each of those axes.
+ *
+ * Finally, the 'offset' tells us where to start the first stripe. In this simple
+ * case the offset is just (0, 0), but in something like a padding operation we
+ * may want to start from a negative index, which is captured by the offset.
```

Comment:
A note about how negative indexing is handled for an operation would be helpful. For example, the stripe config between two operations Op1 and Op2, will depend on the padding needed by Op2, but will influence where Op1 writes its memory. Is that correct?

Reply:
I've updated the doc to use slice as the example here because I think that's easier to follow. Regarding the padding case, it's a bit of a challenge to explain without a diagram but I'll give it a go here.

Let's say we have an op A that represents a symmetric pad by 1 and two tensors T_in and T_out such that T_in -> A -> T_out. If T_in has shape (4, 4) then T_out will have shape (6, 6) after the padding. Now, we choose a StripeConfig for T_out which is equivalent to (2, 2) tiling:

```
StripeConfig T_out = 
{
  shape=[2, 2],
  extent=[6, 6],
  strides=[2, 2],
  order=[1, 2],
  stripes=[3, 3],
  offset=[0, 0],
}
```

The question then is, what StripeConfig will need to be produced when we propagate to T_in? Well, if we state that any part of a stripe that lies outside of the tensor bounds can be ignored, then the input StripeConfig should be this:

```
StripeConfig T_in =
{
  shape=[2, 2],
  extent=[6, 6],
  strides=[2, 2],
  order=[1, 2],
  stripes=[3, 3],
  offset=[-1, -1],
}
```

This will effectively 'overlay' the output StripeConfig on the (4, 4) input tensor T_in, but such that outer 1-wide padding margin is always out-of-bounds (either <0 or >=4 in either axis). This way, reads will never be generated for the padding but they will for the 'interior'.

If this is still unclear - which I appreciate it might be - I can potentially produce a quick diagram.